Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -67,7 +68,8 @@ public final class RegistryClient: Cancellable {
         authorizationProvider: AuthorizationProvider? = .none,
         customHTTPClient: LegacyHTTPClient? = .none,
         customArchiverProvider: ((FileSystem) -> Archiver)? = .none,
-        delegate: Delegate?
+        delegate: Delegate?,
+        checksumAlgorithm: HashAlgorithm
```

Comment:
This moves passing of `checksumAlgorithm` from `downloadSourceArchive` to "global"